how-to/how-to-import-kata-logs-with-fluentd.md

```diff
@@ -0,0 +1,604 @@
+          "Protocol": 2,
+          "Priority": 256,
+          "Table": 254,
+          "Type": 1,
+          "Tos": 0,
+          "Flags": 0,
+          "MPLSDst": null,
+          "NewDst": null,
+          "Encap": null,
+          "MTU": 0,
+          "AdvMSS": 0,
+          "Hoplimit": 0
+        },
+        {
+          "LinkIndex": 4,
+          "ILinkIndex": 0,
+          "Scope": 0,
+          "Dst": {
+            "IP": "ff00::",
+            "Mask": "/wAAAAAAAAAAAAAAAAAAAA=="
+          },
+          "Src": "",
+          "Gw": "",
+          "MultiPath": null,
+          "Protocol": 3,
+          "Priority": 256,
+          "Table": 254,
+          "Type": 1,
+          "Tos": 0,
+          "Flags": 0,
+          "MPLSDst": null,
+          "NewDst": null,
+          "Encap": null,
+          "MTU": 0,
+          "AdvMSS": 0,
+          "Hoplimit": 0
+        }
+      ],
+      "DNS": {
+        "Servers": null,
+        "Domain": "",
+        "Searches": null,
+        "Options": null
+      }
+    },
+    "EndpointType": "virtual",
+    "PCIAddr": ""
+  },
+  "endpoint-type": "virtual",
+  "level": "info",
+  "msg": "endpoint unmarshalled",
+  "name": "kata-runtime",
+  "pid": 15793,
+  "source": "virtcontainers",
+  "subsystem": "network",
+  "time": "2020-02-17T17:54:07Z"
+}
+```
+
+If these new fields are not required, then a `fluentd`
+[`record_transformer` filter](https://docs.fluentd.org/filter/record_transformer#remove_keys)
+could be used to delete them before they are injected into Elastic.
+
+#### Prefixing all keys
+
+It may be noted above that all the fields are imported with their base native name, such as
+`arch` and `level`. It may be better for data storage and processing if all the fields were
+identifiable as having come from Kata, and avoid namespace clashes with other imports.
+This can be achieved by prefixing all the keys with, say, `kata_`. It appears `fluend` cannot
+do this directy in the input or match phases, but can in the filter/parse phase (as was done
+when processing `logfmt` data for instance).  To achieve this, we can first input the Kata
+JSON data as a single line, and then add the prefix using a JSON filter section:
+
+```
+# Pull in as a single line...
+<source>
+  @type tail
+  path /var/log/kata-runtime.log
+  pos_file /var/log/kata-runtime.pos
+  read_from_head true
+  tag kata-runtime
+  <parse>
+    @type none
+  </parse>
+</source>
+
+<filter kata-runtime>
+  @type parser
+  key_name message
+  # drop the original single line `message` entry
+  reserve_data false
+  inject_key_prefix kata_
+  <parse>
+    @type json
+  </parse>
+</filter>
+```
+
+# Caveats
+
+There are a few caveats that should be noted:
+
+- There is a [known issue](https://github.com/kata-containers/runtime/issues/985) whereby enabling
+  full debug in Kata, particularly enabling agent kernel log messages, can result in corrupt log lines
+  being generated by Kata (due to overlapping multiple output streams).
```

Comment:
Yep - the problem is that the kernel logs "corrupt" the nicely structured Kata logs.